packages/template-retail-react-app/app/components/recommended-products/index.jsx

```diff
@@ -130,10 +146,18 @@ const RecommendedProducts = ({zone, recommender, products, title, shouldFetch, .
         }
     }
 
-    // TODO: DRY this handler when intl provider is available globally
     const removeItemFromWishlist = async (product) => {
         try {
-            await wishlist.removeListItemByProductId(product.productId)
+            const wishlistItem = wishlist?.customerProductListItems?.find(
+                (item) => item.productId === product.productId
+            )
+            await deleteCustomerProductListItem.mutateAsync({
+                parameters: {
+                    customerId,
+                    itemId: wishlistItem.id,
```

Comment:
`wishlistItem` is possibly undefined here.

Reply:
I doubt it would be since this is a removal function, it is likely we are deleting the item that is available in the listItem. The only scenerio it could happen is probably some sort of network error that causes the mismatch data between client and data. Still, it is a good callout. I suppose we should return this func in case the `wishlistItem` is not defined